MediaGalleryRenditions/Test/Integration/Model/GenerateRenditionsTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Test\Integration\Model;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Directory\WriteInterface;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\MediaGalleryRenditionsApi\Api\GenerateRenditionsInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\MediaGalleryRenditions\Model\Config;
+use PHPUnit\Framework\TestCase;
+
+class GenerateRenditionsTest extends TestCase
+{
+    /**
+     * @var GenerateRenditionsInterface
+     */
+    private $generateRenditions;
+
+    /**
+     * @var WriteInterface
+     */
+    private $mediaDirectory;
+
+    /**
+     * @var Config
+     */
+    private $renditionSizeConfig;
+
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    protected function setup(): void
+    {
+        $this->generateRenditions = Bootstrap::getObjectManager()->get(GenerateRenditionsInterface::class);
+        $this->mediaDirectory = Bootstrap::getObjectManager()->get(Filesystem::class)
+            ->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->driver = Bootstrap::getObjectManager()->get(DriverInterface::class);
+        $this->renditionSizeConfig = Bootstrap::getObjectManager()->get(Config::class);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        /** @var WriteInterface $mediaDirectory */
+        $mediaDirectory = Bootstrap::getObjectManager()->get(
+            Filesystem::class
+        )->getDirectoryWrite(
+            DirectoryList::MEDIA
+        );
+        if ($mediaDirectory->isExist($mediaDirectory->getAbsolutePath() . '/.renditions')) {
+            $mediaDirectory->delete($mediaDirectory->getAbsolutePath() . '/.renditions');
+        }
+    }
+
+    /**
+     * @dataProvider renditionsImageProvider
+     *
+     * Test for generation of rendition images.
+     *
+     * @param array $paths
+     * @param string $renditionPath
+     * @param bool $isRenditionsGenerated
+     * @throws LocalizedException
+     */
+    public function testExecute(array $paths, string $renditionPath, bool $isRenditionsGenerated): void
+    {
+        $this->copyImage($paths);
+        $this->generateRenditions->execute($paths);
+        $expectedRenditionPath = $this->mediaDirectory->getAbsolutePath($renditionPath);
+        if ($isRenditionsGenerated) {
```

Comment:
Condition can be simplified:
```suggestion
        if (!$isRenditionsGenerated) {
            $this->assertFileDoesNotExist($expectedRenditionPath);
            return;
        }
```